xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -214,7 +214,7 @@ actor UserProfileService {
     
     /// List all the users followed by the passed sphere.
     /// Each user will be decorated with whether the current app user is following them.
-    private func getFollowingList<Sphere: SphereProtocol>(
+    func getFollowingList<Sphere: SphereProtocol>(
         sphere: Sphere,
```

Comment:
I believe the sphere _must_ belong to the same Noosphere instance. It may be better to pass in a sphere identifier, and use `self.noosphere` to get the sphere instance. That way it is impossible to pass the wrong sphere instance.

Reply:
I had to introduce `NoosphereService.sphere(did: Did) -> Sphere` to actually implement this. Even if you have `NoosphereService` and a `Did` you can't call `Sphere.init()` because the inner `Noosphere` field is private on the service.

I think this solution is fine but just flagging because any change to `NoosphereService` should be considered.